### PR TITLE
Make ``LookupMixIn`` inherit from ``NodeNG``

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] Write a good description on what the PR does.

## Description

Fixing typing issues one message at a time.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Related Issue

### CODE REVIEW

File: astroid/nodes/scoped_nodes/mixin.py [hunk 1]
```diff
@@ -50,6 +50,24 @@ def scope(self: _T) -> _T:
         """
         return self
 
+    def scope_lookup(self, node, name: str, offset: int = 0):
+        """Lookup where the given variable is assigned.
+
+        :param node: The node to look for assignments up to.
+            Any assignments after the given node are ignored.
+        :type node: NodeNG
```
Comment (Pierre-Sassoulas) [inline astroid/nodes/scoped_nodes/mixin.py:58]:
```suggestion
```

File: astroid/nodes/scoped_nodes/mixin.py [hunk 2]
```diff
@@ -50,6 +50,24 @@ def scope(self: _T) -> _T:
         """
         return self
 
+    def scope_lookup(self, node, name: str, offset: int = 0):
```
Comment (Pierre-Sassoulas) [inline astroid/nodes/scoped_nodes/mixin.py:53]:
```suggestion
    def scope_lookup(self, node: NodeNG, name: str, offset: int = 0) -> tuple(str, list(NodeNG):
```

File: astroid/nodes/scoped_nodes/mixin.py [hunk 3]
```diff
@@ -50,6 +50,24 @@ def scope(self: _T) -> _T:
...
+        """Lookup where the given variable is assigned.
+
+        :param node: The node to look for assignments up to.
+            Any assignments after the given node are ignored.
+        :type node: NodeNG
+
+        :param name: The name of the variable to find assignments for.
+
+        :param offset: The line offset to filter statements up to.
+
```
Comment (Pierre-Sassoulas) [inline astroid/nodes/scoped_nodes/mixin.py:63]:
```suggestion
        :param offset: The line offset to filter statements up to.
```

File: astroid/nodes/scoped_nodes/mixin.py [hunk 4]
```diff
@@ -50,6 +50,24 @@ def scope(self: _T) -> _T:
...
+        :returns: This scope node and the list of assignments associated to the
+            given name according to the scope where it has been found (locals,
+            globals or builtin).
+        :rtype: tuple(str, list(NodeNG))
```
Comment (Pierre-Sassoulas) [inline astroid/nodes/scoped_nodes/mixin.py:67]:
```suggestion
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
I guess it make sense to make the mixin inherit from the base class for node as they will always be used in a child class of it.

File: astroid/nodes/scoped_nodes/mixin.py (hunk 2 above)
Comment (DanielNoord) [reply astroid/nodes/scoped_nodes/mixin.py:53]: I'm not sure this is actually true. Wherever `node: NodeNG` I have become real suspicious that it should be `node: NodeNG | Instance`.
I'll investigate it later, `mypy` at least passes now.

That's also why I let the docstring in.

File: astroid/nodes/scoped_nodes/mixin.py [hunk 5]
```diff
@@ -50,6 +50,24 @@ def scope(self: _T) -> _T:
...
+        """
+        raise NotImplementedError
```
Comment (Pierre-Sassoulas) [inline astroid/nodes/scoped_nodes/mixin.py:69]:
```suggestion
        raise NotImplementedError  # pragma: no cover
```

**Review by Pierre-Sassoulas — APPROVED** [review]

File: astroid/nodes/scoped_nodes/mixin.py (hunk 5 above)
Comment (DanielNoord) [reply astroid/nodes/scoped_nodes/mixin.py:69]: I'll add this to the last of standard uncovered lines.

**Pull request merged**